code/game/atoms_movable.dm

```diff
@@ -456,6 +459,14 @@
 			return
 	A.Bumped(src)
 
+///Sets the anchored var and returns if it was sucessfully changed or not.
+/atom/movable/proc/setAnchored(anchorvalue)
+	if(anchored == anchorvalue)
+		return FALSE
+	anchored = anchorvalue
+	SEND_SIGNAL(src, COMSIG_MOVABLE_SETANCHORED, anchorvalue)
+	return TRUE
```

Comment:
```suggestion
	SHOULD_CALL_PARENT(TRUE)
	if(anchored == anchorvalue)
		return
	. = anchored
	anchored = anchorvalue
	SEND_SIGNAL(src, COMSIG_MOVABLE_SETANCHORED, anchorvalue)
```
I tend to use this pattern as things can end up with more values than TRUE/FALSE.
You can still check if there was no change with `isnull(.)` on proc redefinitions.

Also ninjanomnom tends to prefer signals to run as early as possible. What do you think of making it run before the value is changed here, to allow for an early return later on, in case something wants to block an anchoring change?
That might require changing existing usages of the signal.